src/Common/EnumerableExtensions.cs

```diff
@@ -10,12 +10,6 @@ internal static class EnumerableExtensions
         return source.DefaultIfNotSingle(static _ => true);
     }
 
-    /// <inheritdoc cref="DefaultIfNotSingle{TSource}(ImmutableArray{TSource}, Func{TSource, bool})"/>
-    public static TSource? DefaultIfNotSingle<TSource>(this ImmutableArray<TSource> source)
```

Comment:
These are here for symmetry with the `IEnumerable` version. It's OK to remove them if they aren't used, as long as you're also aware that this is highlighting a boxing that we probably should fix elsewhere.

Reply:
There's only one instance where this is used in production code in AsShouldBeUsedOnlyForInterfaceAnalyzer, which is a LINQ projection and uses IEnumerable<T>

Reply:
For the boxing, the ECS analyzer should fire when we have a new boxing issue introduced.